examples/showcase/content/docs/use-server-action.mdx

```diff
@@ -144,6 +144,7 @@ function MyComponent() {
 - `execute`: A function to execute the server action with the provided input.
 - `setOptimistic`: A function to set an optimistic update for the `data` value. It accepts either the new data directly or a function that receives the current data and returns the new data.
 - `reset`: A function to reset the state of the server action to its initial values.
+- `lastUsedInput`: The last input evaluated by the server action to get to the current state. This value will be the input before parsing and validation. If the input is FormData, it will be represented as a JSON object.
```

Comment:
If I'm understanding correctly this solution has the same problem as described here no?

https://github.com/IdoPesok/zsa/pull/124#issuecomment-2184950717


Why returning data when error is not a good idea? I would like to understand. 

Traditional server techs like Rails or others validate a form and you can access old data in the fields that didn't fail. 

And the important part is you always access data, in the same way, it doesn't matter if there was an error.

Reply:
Good question. I think we have a different definition of `data`. The way I define it, and the way its represented in `zsa` docs, is that it is the return value of a successful handler. What you returned in your PR is not data, rather it is what I define as  `input`. Input is the value passed to the server action.

- `data`: the successful return value of a server action's handler
- `input`: the value sent to the server action over the network
- `error`: a thrown error in a server action

So for example, if we have a simple action

```typescript
export const simpleAction = createServerAction()
  .input(
    z.object({
      name: z.string(),
    })
  )
  .handler(async ({ input }) => {
    return `Hello, ${input.name}`
  })
```

Applied definitions:

- `data`: `Hello, ${name}`
- `input`: { name: "..." }
- `error`: a thrown error in a server action

ZSA server actions must follow the following type:

```
(input) => [data, null] | [null, err]
```

I think we are getting at the same thing, but what you are looking for is what I define as `input` and you define as `data`.

Rather than doing `const { data } = useServerAction(...)`, you will do `const { input } = useServerAction(...)`. From the open source code you posted, I see your function doesn't have `data` because it is doing a redirect.

Hopefully that makes sense!

Reply:
It makes totally sense, thanks for the detailed response!

looking forward to try this one. ping me when you have ready and I'll be the first Beta tester :)